lxd/device/disk.go

```diff
@@ -157,6 +157,40 @@ func (d *disk) sourceIsLocalPath(source string) bool {
 	return true
 }
 
+// Check that unshared custom storage block volumes are not added to profiles or multiple instances.
+func (d *disk) checkBlockVolSharing(instanceType instancetype.Type, projectName string, volume *api.StorageVolume) error {
+	contentType, err := storagePools.VolumeContentTypeNameToContentType(volume.ContentType)
+	if err != nil {
+		return err
+	}
+
+	// Skip the checks if the volume is set to be shared or is not a block volume.
+	if contentType != cluster.StoragePoolVolumeContentTypeBlock || shared.IsTrue(volume.Config["security.shared"]) {
+		return nil
+	}
+
+	if instanceType == instancetype.Any {
+		return fmt.Errorf("Cannot add custom storage block volume to profile if security.shared is false or unset")
+	}
+
+	usedByInstance := false
+
+	err = storagePools.VolumeUsedByInstanceDevices(d.state, d.pool.Name(), projectName, volume, true, func(inst db.InstanceArgs, project api.Project, usedByDevices []string) error {
+		usedByInstance = true
+
+		return db.ErrListStop
+	})
+	if err != nil && err != db.ErrListStop {
```

Comment:
Rather than having a usedByInstance var and a separate if statement below, we can do this more concisely like we do in `UsedBy`:

```go
	if err != nil {
		if err == db.ErrListStop  {
			return fmt.Errorf("Cannot add custom storage block volume to more than one instance nor to profiles if security.shared is false or unset")
		}

		return err
	}
```